src/app/cake/cake.model.ts

```diff
@@ -0,0 +1,32 @@
+export interface CakeInterface {
+	id: number
+	name: string
+  descrition: string
+  ingredients: string[]
+  price: number
+  stock: number
+  status: 'Available'|'LastUnits'|'OutOfStock'
+
+	newCake(
+		id: number,
+		newCake: Cake
+	): Cake
+
+	getCakes(): Cake[]
+
+	editCake(id: number): Cake
+
+	updateCake(id: number, newCakeData: Cake): Cake
+
+	deleteCake(id: number): void
+}
+
+export type Cake = {
+  id: number
+	name: string
+  descrition: string
+  ingredients: string[]
+  price: number
+  stock: number
+  status: 'Available'|'LastUnits'|'OutOfStock'
```

Comment:
You should create an enumerator for this. 

Since we are using a Single File Responsibility file distribution, this Enumerator should live in a separate file named `cake-status.enum.ts`.